src/models/image-map.ts

```diff
@@ -204,38 +250,70 @@ export const ImageMapModel = types
+                success: true
               };
-        return self.addImage(entry.contentUrl!, entry);
+        return self._addImage(entry.contentUrl!, entry);
       }),
 
-      getImage: flow(function* (url: string, options?: IImageBaseOptions)
+      _storeAndAddImage: flow(function* (url: string, handler: IImageHandler, options?: IImageBaseOptions)
                        : Generator<PromiseLike<any>, ImageMapEntryType | Promise<ImageMapEntryType>, unknown> {
-        if (!url) {
-          return clone(self.images.get(placeholderImage)!);
-        }
+        const storeResult = (yield handler.store(url, { db: _db, ...options })) as IImageHandlerStoreResult;
+        return self._addImage(url, storeResult);
+      }),
+    };
+  })
+  .actions(self => ({
+    // eslint-disable-next-line require-yield
+    getImage: flow(function* (url: string, options?: IImageBaseOptions)
+                    : Generator<PromiseLike<any>, ImageMapEntryType | Promise<ImageMapEntryType>, unknown> {
+      if (!url) {
+        // TODO: how often does this happen, should it be silently ignored like this?
+        return clone(self.images.get(placeholderImage)!);
+      }
 
-        const imageEntry = self.images.get(url);
-        if (imageEntry) {
-          // This might or might not have dimensions yet
-          return imageEntry;
-        }
+      const imageEntry = self.images.get(url);
+      if (imageEntry?.status === EntryStatus.Ready) {
+        // This image has been downloaded successfully already
+        return imageEntry;
+      }
 
-        const handler = self.getHandler(url);
-        if (handler) {
-          const imageEntrySnapshot = (yield handler.store(url, { db: _db, ...options })) as ImageMapEntrySnapshot;
-          return self.addImage(url, imageEntrySnapshot);
-        }
-        else {
-          return clone(self.images.get(placeholderImage)!);
-        }
-      })
+      const existingStoringPromise = self.storingPromises[url];      
+      if (existingStoringPromise && imageEntry?.status !== EntryStatus.Error) {
+        // If the imageEntry is errored we ignore the existing promise
+        // This way a second getImage request will try to store the image again
+        return existingStoringPromise;
+      }
 
-    };
-  });
+      const handler = self.getHandler(url);
+      if (!handler) {
+        console.warn(`No handler found for ${url}`);
+        return clone(self.images.get(placeholderImage)!);
+      }
+
+      // If there is an existing entry we'll overwrite it so it's status is
+      // storing and the displayUrl is the placeholder. In theory the
+      // existingEntry could have a status of Storing, ComputingDimensions, or
+      // Error. Because there is no existingStoringPromise the status should
+      // really not be Storing or ComputingDimensions.
+      if (imageEntry?.status === EntryStatus.Storing || 
+          imageEntry?.status === EntryStatus.ComputingDimensions) {
```

Comment:
```suggestion
      // Error. Because there is no existingStoringPromise the status should
      // really not be PendingStorage or PendingDimensions.
      if (imageEntry?.status === EntryStatus.PendingStorage || 
          imageEntry?.status === EntryStatus. PendingDimensions) {
```